common/persistence/dataStoreInterfaces.go

```diff
@@ -349,7 +349,8 @@ type (
 		SignalRequestedIDs  map[string]struct{}
 		BufferedEvents      []*DataBlob
 
-		Checksum checksum.Checksum
+		Checksum     checksum.Checksum
+		ChecksumData *DataBlob
```

Comment:
Can you please add comments on these fields and explain why we need these 2. At first look it's confusing